src/dotty/tools/dotc/sbt/ThunkHolder.scala

```diff
@@ -0,0 +1,59 @@
+package dotty.tools.dotc
+package sbt
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+import xsbti.api
+
+/** Create and hold thunks */
```

Comment:
This trait is very small and used only by `ExtractAPICollector`. It's not clear what it does. I propose to merge it into `ExtractAPICollector`